docs/instructor-training-workflow.txt

```diff
@@ -0,0 +1,249 @@
+  - Trainee views {individual training progress}.
+  - Trainee enters URL of written work.
+  - AMY updates <Training Task> associated with <Person> and <Event>.
+  - AMY marks <Training Progress> as "Needs Review".
+
+* [admin reports submission of written work]: on behalf of trainee.
+  - Admin looks up <Training Progress> associated with trainee.
+  - Admin enters URL of written work in <Training Task>.
+  - AMY marks <Training Progress> as "Needs Review".
+
+* [admin reports completion of written work]: on behalf of trainee.
+  - Admin looks up <Training Progress> associated with trainee.
+  - Admin enters URL of written work in <Training Task>.
+  - Admin marks <Training Task> as complete.
+
+## Tracking participation in interactive work.
+
+* [admin reports individual completion of interactive work]: such as discussion session or lesson demo.
+  - Admin views {individual training progress} for trainee.
+  - Admin selects <Training Task>.
+  - Admin:
+    - Enters date of interactive event.
+    - Adds comments (if any).
+  - Admin indicates whether <Training Task> is now complete.
+
+* [admin reports bulk completion of interactive work]: for all participants.
+  - Admin views {training task completion}.
+  - Admin selects one or more <Person>s.
+  - Admin selects <Training Task> (e.g., discussion, demo).
+  - AMY marks all <Training Task>s as "Completed".
+
+## Completing training work.
+
+* [admin changes completion status of work for trainee]
+  - Admin looks up <Training Progress> associated with trainee.
+  - Admin changes "Completed" status of work item(s).
+  > [training automatically completed]
+
+* [admin finds work to review]
+  - Admin views {training progress summary}.
+  - Admin selects <Training Progress>.
+  > [admin changes completion status of work for trainee].
+
+## Completing training.
+
+* [training automatically completed]: triggered by changes in <Training Task>s.
+  - AMY records change to <Training Task> and all <Training Task>s for <Person> now complete.
+  - AMY displays {badge creation form}.
+  - Admin confirms creation of badge.
+  - AMY marks <Training Progress> as "Completed".
+  - AMY creates certificate.
+  - AMY mails certificate to <Person>.
+
+## Ending training.
+
+* [admin ends training]: because trainee has gone silent, failed a work item, etc.
+  - Admin views {individual training progress}.
+  - Admin selects <Training Progress>.
+  - Admin marks <Training Progress> as "Unsuccessful".
+  - AMY marks <Training Progress> as "Completed".
+
+* [admin ends training course]: because it has timed out.
+  - Admin selects training <Event>.
+  - Admin marks <Event> as "Stalled".
+  - AMY marks all associated <Training Progress>es as "Unsuccessful".
+  - AMY marks <Training Progress> as "Completed".
+
+## Questions
+
+- What to do if Admin deletes <Task> from training <Event> directly rather than via [admin removes people from training event]?
```

Comment:
OK, so this looks like redundancy - you have Tasks indicating that a Person was matched with an Event, and at the same time, you can match Application to an Event. What if we drop `TrainingApplication.event` and rely only on Tasks records?